python/cudf/cudf/core/dataframe.py

```diff
@@ -7574,7 +7574,7 @@ def to_struct(self, name=None):
             names=self._data.names, children=self._data.columns, size=len(self)
         )
         return cudf.Series._from_data(
-            cudf.core.column_accessor.ColumnAccessor({name: col}),
+            cudf.core.column_accessor.ColumnAccessor({name: col.copy()}),
```

Comment:
A question related to the one about the `copy` API above, but perhaps more pertinent in the context of this PR: are there any other cases where we're not copying but we should be?

Reply:
No obvious places come to mind. Did a quick check on `rename` and `reset_index` (as those are referenced in the Pandas PR) and those look fine.